### PR TITLE
Add COLLABORATOR_GUIDE.md with section on releases

### PR DESCRIPTION
**Pull request by gesinger**
## Requirements Checklist
- [ ] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [ ] Unit Tests updated or fixed
  - [X] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](https://jsbin.com/gejugat/edit?html,output))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: COLLABORATOR_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+#### GitHub personal access token
+
+This is used to make a GitHub release on videojs. You can get a token from the [personal access tokens](https://github.com/settings/tokens) page.
+
+After generating one, make sure to keep it safe because GitHub will not show the token for you again. A good place to save it is Lastpass Secure Notes.
+
+### Deciding what type of version release
+
+Since we follow the [conventional changelog conventions][conventions], all commits are prepended with a type, most commonly `feat` and `fix`.
```
Comment (ldayananda) [inline COLLABORATOR_GUIDE.md]: forgot the link for `conventions`?

File: COLLABORATOR_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+If any commit has a `BREAKING CHANGE` footer, then the release will be a `major` release.
+Most common releases will be either `patch` or `minor`.
+
+### Doing a release
+
+To make the release process easier, and in case you have a dirty repo from development, it is recommended that you checkout a clean clone of http-streaming.
+
+```sh
+git clone git@github.com:videojs/http-streaming vhs-release
+cd vhs-release
```
Comment (ldayananda) [inline COLLABORATOR_GUIDE.md]: maybe include `nvm use` here?

File: COLLABORATOR_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+Depending on the commits that have been merged, you can choose from `major`, `minor`, or `patch` as the versioning values.
+See [deciding what type of version release section](#deciding-what-type-of-version-release).
+
+Optionally, you can run `git show` now to verify that the version update and CHANGELOG automation worked as expected.
+
+Afterwards, you want to push the commit and the tag to the repo.
+It's necessary to do this before running `npm publish` because our GitHub release automation relies on the commit being available on GitHub.
+
+```sh
+git push --tags origin master
```
Comment (ldayananda) [inline COLLABORATOR_GUIDE.md]: not the actual tag number?

File: COLLABORATOR_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+Run `npm publish` with an appropriate tag. Don't forget to supply your token.
+
+```sh
+VJS_GITHUB_USER=your-github-username VJS_GITHUB_TOKEN=my-personal-access-token npm publish --tag next
```
Comment (ldayananda) [inline COLLABORATOR_GUIDE.md]: I don't think this command is applicable to us. Just
`npm publish` should be enough

File: COLLABORATOR_GUIDE.md (hunk 3 above)
Comment (gkatsev) [reply COLLABORATOR_GUIDE.md]: at least --follow-tags, as we discussed in slack

File: COLLABORATOR_GUIDE.md (hunk 4 above)
Comment (gkatsev) [reply COLLABORATOR_GUIDE.md]: [It *could* :)](https://github.com/videojs/video.js/blob/master/build/gh-release.js)

File: COLLABORATOR_GUIDE.md (hunk 1 above)
Comment (gkatsev) [reply COLLABORATOR_GUIDE.md]: Here's a handy link: https://github.com/videojs/conventional-changelog-videojs/blob/master/convention.md

File: COLLABORATOR_GUIDE.md (hunk 2 above)
Comment (gesinger) [reply COLLABORATOR_GUIDE.md]: Thanks! Had it in the other file and forgot to migrate it over.

File: COLLABORATOR_GUIDE.md (hunk 3 above)
Comment (gkatsev) [reply COLLABORATOR_GUIDE.md]:
```sh
$ git push origin v`cat package.json | grep version | head -n 1 | cut -f 4 -d \"`
```

File: COLLABORATOR_GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+git push --follow-tags origin master
+```
+
+Publish to npm.
+
+```sh
+npm publish
+```
+
+After it's done, [create a release in github](https://github.com/videojs/http-streaming/releases/new) with latest tag, no title or description, and attach the .min.js and .js dist files.
```
Comment (gkatsev) [inline COLLABORATOR_GUIDE.md]: would be nice if the changelog for the version was copied into the release text as well

**Pull request merged**